src/structures/message/Message.ts

```diff
@@ -0,0 +1,133 @@
+	public content!: string;
+	public createdTimestamp!: number;
+	public createdAt!: Date;
+	public editedTimestamp?: number;
+	public editedAt?: Date;
+	public tts!: boolean;
+	/* public mentionEveryone!: boolean;
+	public mentions!: (User & {
+		member?: Omit<GuildMember, 'user'>;
+	})[];
+	public mentionRoles!: Snowflake[];
+	public mentionChannels?: APIChannelMention[];*/
+	public attachments!: MessageAttachment[];
+	public embeds!: Embed[];
+	public reactions?: APIReaction[];
+	public nonce?: string | number;
+	public pinned!: boolean;
+	public webhookId?: Snowflake;
+	public type!: MessageType;
+	public activity?: MessageActivity;
+	public application?: Partial<Application>;
+	public messageReference?: MessageReference;
+	public flags?: MessageFlags;
+	public stickers?: Sticker[];
+	public referencedMessage?: Message;
+	public interaction?: MessageInteraction;
+
+	public constructor(public readonly channel: GuildChannel, data: APIMessage) {
+		this.client = channel.client;
+
+		this.$patch(data);
+	}
+
+	public $patch(data: APIMessage): void {
+		this.id = data.id;
+		this.channelId = data.channel_id;
+		this.guildId = data.guild_id;
+		this.author = new User(this.client, data.author);
+		this.content = data.content;
+		this.createdTimestamp = Number(data.timestamp);
+		this.createdAt = new Date(this.createdTimestamp);
+		this.editedTimestamp = Number(data.edited_timestamp);
+		this.editedAt = this.editedTimestamp ? new Date(this.editedTimestamp) : undefined;
+		this.tts = data.tts;
+		/*this.mentionEveryone = data.mention_everyone;
+		this.mentions = data.mentions;
+		this.mentionRoles = data.mention_roles;
+		this.mentionChannels = data.mention_channels;*/
+		this.attachments = data.attachments.map((attachment) => new MessageAttachment(attachment));
+		this.embeds = data.embeds.map((embed) => new Embed(embed));
+		this.reactions = data.reactions;
+		this.nonce = data.nonce;
+		this.pinned = data.pinned;
+		this.webhookId = data.webhook_id;
+		this.type = data.type;
+		this.activity = data.activity && {
+			type: data.activity.type,
+			partyId: data.activity.party_id,
+		};
+		this.application = data.application && new Application(this.client, data.application);
+		this.messageReference = data.message_reference && {
+			messageId: data.message_reference.message_id,
+			channelId: data.message_reference.channel_id,
+			guildId: data.message_reference.guild_id,
+		};
+		this.flags = data.flags;
+		this.stickers = data.stickers?.map((sticker) => new Sticker(this.client, sticker));
+		this.referencedMessage = data.referenced_message ? new Message(this.channel, data.referenced_message) : undefined;
+		this.interaction = data.interaction && {
+			...data.interaction,
+			user: new User(this.client, data.interaction.user),
+		};
+	}
+
+	public get guild(): Guild | undefined {
+		return this.channel.guild;
+	}
+
+	public get member(): GuildMember | undefined {
+		return this.guild ? this.guild.members.get(this.author.id) : undefined;
+	}
```

Comment:
vars

Reply:
I don't understand

Reply:
guild should not be a getter

Reply:
Yes, it should, I just messed up the parameters of the constructor, it doesn't take a GuildChannel because the message can be sent in DM. So we should retrieve it using guildId using the future cache system.